docs/dev/2026-05-15-rag-feature.md

```
@@ -0,0 +1,473 @@
+    }
+    registry.Register(tools.NewRAGSearchTool(ragClient, cfg.RAG))
+}
+```
+
+### Step 5: Implement the tool
+
+Create `pkg/tools/rag_search.go`. Tool name: `rag_search`.
+
+Tool description for the LLM:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)